### PR TITLE
update typescript to 4.7

### PR DESCRIPTION
**Pull request by mattcasey**
Released May 24: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/

### CODE REVIEW

File: lib/browser.ts [hunk 1]
```diff
@@ -30,8 +30,7 @@ export function isTouchDevice () {
   }
   else {
     const mQ = typeof window !== 'undefined'
-      && window.matchMedia
-      && matchMedia('(pointer:coarse)');
+      && window.matchMedia('(pointer:coarse)');
```
Comment (mattcasey) [inline lib/browser.ts:33]: This is a code change but should be OK. According to https://caniuse.com/matchmedia, every browser supports it

File: .vscode/settings.json [hunk 2]
```diff
@@ -20,5 +20,6 @@
   "editor.codeActionsOnSave": {
     "source.organizeImports": false
   },
-  "jestrunner.configPath": "jest.config-server.ts"
+  "jestrunner.configPath": "jest.config-server.ts",
+  "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (mattcasey) [inline .vscode/settings.json:24]: make VS Code use the correct version

Comment (Devorein) [reply .vscode/settings.json:24]: Nice

**Review by Devorein — APPROVED** [review]

File: lib/browser.ts (hunk 1 above)
Comment (Devorein) [reply lib/browser.ts:33]: Nice. good on ts for detecting a condition that will always be truthy

**Pull request merged**